src/ui/components/util/Localization.ts

```diff
@@ -0,0 +1,77 @@
+import type { PlayerError, VideoQuality } from 'react-native-theoplayer';
+
+export type Localization = Partial<AllLocalization>;
+
+export interface AllLocalization {
+  backButton: string;
+  settingsTitle: string;
+  qualityTitle: string;
+  qualityLabel: ({ quality }: { quality: VideoQuality | undefined }) => string;
+  qualityLabelExtended: ({ quality }: { quality: VideoQuality | undefined }) => string;
+  audioTitle: string;
+  subtitleTitle: string;
+  playbackRateTitle: string;
+  playbackRateValue: ({ rate }: { rate: number }) => string;
+  adLabel: ({ currentAd, totalAds }: { currentAd: number; totalAds: number }) => string;
+  adSkipLabel: ({ seconds }: { seconds: number }) => string;
+  adCountdown: ({ remainingDuration }: { remainingDuration: number }) => string;
+  adClickThroughButton: string;
+  castMessage: ({ state, target }: { state: 'connecting' | 'connected'; target: 'Chromecast' | 'Airplay' }) => string;
+  errorMessage: ({ error }: { error: PlayerError }) => string;
+  liveLabel: string;
+}
+
+export const defaultLocalization: AllLocalization = {
+  backButton: 'Back',
+  settingsTitle: 'Settings',
+  qualityTitle: 'Video Quality',
+  audioTitle: 'Audio',
+  subtitleTitle: 'Subtitles',
+  playbackRateTitle: 'Playback Speed',
+  liveLabel: 'LIVE',
+  qualityLabel: ({ quality }) => {
+    if (quality === undefined) return 'auto';
+    if (quality.label && quality.label !== '') return quality.label;
+    if (quality.height) return quality.height + 'p';
+    return '';
+  },
+  qualityLabelExtended: ({ quality }) => {
```

Comment:
There's a bit too much complexity in this function. Consider splitting this up into more focused localization keys, and moving more of this logic back into the UI.